cluster-operator/src/main/java/io/strimzi/operator/cluster/model/cruisecontrol/CapacityConfiguration.java

```diff
@@ -0,0 +1,257 @@
+        Map<Integer, BrokerCapacityOverride> overrideMap = new HashMap<>();
+
+        if (brokerCapacity != null && brokerCapacity.getOverrides() != null && !brokerCapacity.getOverrides().isEmpty()) {
+            for (BrokerCapacityOverride override : brokerCapacity.getOverrides()) {
+                List<Integer> ids = override.getBrokers();
+
+                for (int id : ids) {
+                    if (overrideMap.containsKey(id)) {
+                        LOGGER.warnCr(reconciliation, "Duplicate broker id {} found in overrides, using first occurrence.", id);
+                    } else if (kafkaBrokerNodes.stream().noneMatch(node -> node.nodeId() == id)) {
+                        LOGGER.warnCr(reconciliation, "Ignoring broker capacity override for unknown node ID {}", id);
+                    } else {
+                        overrideMap.put(id, override);
+                    }
+                }
+            }
+        }
+
+        return overrideMap;
+    }
+
+    private static TreeMap<Integer, CapacityEntry> generateCapacityEntries(Reconciliation reconciliation,
+                                         CruiseControlSpec spec,
+                                         Set<NodeRef> kafkaBrokerNodes,
+                                         Map<String, Storage> kafkaStorage,
+                                         Map<String, ResourceRequirements> kafkaBrokerResources) {
+        TreeMap<Integer, CapacityEntry> capacityEntries = new TreeMap<>();
+        BrokerCapacity generalBrokerCapacity = spec.getBrokerCapacity();
+        Map<Integer, BrokerCapacityOverride> brokerCapacityOverrideMap = processBrokerCapacityOverrides(reconciliation,
+                kafkaBrokerNodes, generalBrokerCapacity);
+
+        for (NodeRef node : kafkaBrokerNodes) {
+            BrokerCapacityOverride brokerCapacityOverride = brokerCapacityOverrideMap.get(node.nodeId());
+
+            DiskCapacity disk = new DiskCapacity(kafkaStorage.get(node.poolName()), node.nodeId());
+            CpuCapacity cpu = new CpuCapacity(generalBrokerCapacity, brokerCapacityOverride, kafkaBrokerResources.get(node.poolName()));
+            InboundNetworkCapacity inboundNetwork = new InboundNetworkCapacity(generalBrokerCapacity, brokerCapacityOverride);
+            OutboundNetworkCapacity outboundNetwork = new OutboundNetworkCapacity(generalBrokerCapacity, brokerCapacityOverride);
+
+            CapacityEntry capacityEntry = new CapacityEntry(node.nodeId(), disk, cpu, inboundNetwork, outboundNetwork);
+            capacityEntries.put(node.nodeId(), capacityEntry);
+        }
+
+        return capacityEntries;
+    }
+
+    /**
+     * Generate a capacity configuration for cluster.
+     *
+     * @return Cruise Control capacity configuration as a formatted JSON String.
+     */
+    public String toJson() {
+        JsonArray capacityList = new JsonArray();
+
+        for (CapacityEntry capacityEntry : capacityEntries.values()) {
```

Comment:
Is it possible that the collection is empty? What would be the JSON in that case? Would it be valid for Cruise Control? Maybe an edge case which is not likely to happen?

Reply:
The JSON would be empty in that case which is an invalid configuration for Cruise Control. That being said, it could only happen if we instantiated the `CapacityConfiguration` class with an invalid list of broker nodes `kafkaBrokerNodes`. 

Do you think this is something I should add a unit test for or some warning?

Reply:
Thanks Kyle!